components/match2/wikis/leagueoflegends/match_group_input_custom.lua

```diff
@@ -155,425 +144,286 @@ end
+			match = MatchGroupInput.readPlayersOfTeam(match, opponentIndex, opponent.name, {
+				resolveRedirect = true,
+				applyUnderScores = true,
+				maxNumPlayers = MAX_NUM_PLAYERS,
+			})
+		elseif opponent.type == Opponent.solo then
+			opponent.match2players = Json.parseIfString(opponent.match2players) or {}
+			opponent.match2players[1].name = opponent.name
 		end
+
+		return opponent
+	end)
+end
+
+---@param scoreInput string|number|nil
+---@return integer? #SCORE
+---@return string? #STATUS
+function MatchFunctions._parseScoreInput(scoreInput)
+	if not scoreInput then
+		return
 	end
 
-	--apply walkover input
-	match.walkover = string.upper(match.walkover or '')
-	if Logic.isNumeric(match.walkover) then
-		local winnerIndex = tonumber(match.walkover)
-		opponents = MatchFunctions._makeAllOpponentsLoseByWalkover(opponents, STATUS_DEFAULT_LOSS)
-		opponents[winnerIndex].status = STATUS_DEFAULT_WIN
-		match.finished = true
-	elseif Logic.isNumeric(match.winner) and Table.includes(ALLOWED_STATUSES, match.walkover) then
-		local winnerIndex = tonumber(match.winner)
-		opponents = MatchFunctions._makeAllOpponentsLoseByWalkover(opponents, match.walkover)
-		opponents[winnerIndex].status = STATUS_DEFAULT_WIN
-		match.finished = true
+	if Logic.isNumeric(scoreInput) then
+		return tonumber(scoreInput), MatchGroupInput.STATUS.SCORE
 	end
 
-	-- see if match should actually be finished if score is set
-	if not Logic.readBool(match.finished) then
-		MatchFunctions._finishMatch(match, opponents, isScoreSet)
+	local scoreUpperCase = string.upper(scoreInput)
+	if Table.includes(MatchGroupInput.STATUS_INPUTS, scoreUpperCase) then
+		return MatchGroupInput.SCORE_NOT_PLAYED, scoreUpperCase
 	end
+end
 
-	-- apply placements and winner if finshed
-	if Logic.readBool(match.finished) and not match.standaloneMatch then
-		match, opponents = CustomMatchGroupInput.getResultTypeAndWinner(match, opponents)
+---@param walkoverInput string #wikicode input
+---@param isWinner boolean
+---@return integer? #SCORE
+---@return string? #STATUS
+function MatchFunctions._opponentWalkover(walkoverInput, isWinner)
+	if Logic.isNumeric(walkoverInput) then
+		walkoverInput = MatchGroupInput.STATUS.DEFAULT_LOSS
 	end
 
-	-- Update all opponents with new values
-	for opponentIndex, opponent in pairs(opponents) do
-		match['opponent' .. opponentIndex] = opponent
+	local walkoverUpperCase = string.upper(walkoverInput)
+	if Table.includes(MatchGroupInput.STATUS_INPUTS, walkoverUpperCase) then
+		return MatchGroupInput.SCORE_NOT_PLAYED, isWinner and MatchGroupInput.STATUS.DEFAULT_WIN or walkoverUpperCase
 	end
-	return match
 end
 
 ---@param match table
----@param opponents table[]
----@param isScoreSet boolean
----@return table
-function MatchFunctions._finishMatch(match, opponents, isScoreSet)
+---@param opponents {score: integer?}[]
+---@return boolean
+function MatchFunctions._isFinished(match, opponents)
+	if Logic.readBool(match.finished) then
```

Comment:
maybe readBoolOrNil ~= nil and return the bool then (even if false)
that would allow manual input to say that a match isn't finished